### PR TITLE
[improve] configure whether function consumer should skip to latest

### PR DESCRIPTION
**Pull request by nlu90**
*(If this PR fixes a github issue, please add `Fixes #<xyz>`.)*

Fixes #<xyz>

*(or if this PR is one task of a github issue, please add `Master Issue: #<xyz>` to link to the master issue.)*

Master Issue: #<xyz>

### Motivation

In certain failure cases, the function needs to skip all the content between the last successfully Acked message and the latest message in the topic in order for quick recovery. 

### Modifications

1. Providing a boolean config for function submission cmd
2. PulsarSource will call `consumer.seek(MessageId.latest)` if the skip flag is set

### Verifying this change

- [ ] Make sure that the change passes the CI checks.

*(Please pick either of the following options)*

This change is a trivial rework / code cleanup without any test coverage.

*(or)*

This change is already covered by existing tests, such as *(please describe tests)*.

*(or)*

This change added tests and can be verified as follows:

*(example:)*
  - *Added integration tests for end-to-end deployment with large payloads (10MB)*
  - *Extended integration test for recovery after broker failure*

### Does this pull request potentially affect one of the following parts:

*If `yes` was chosen, please highlight the changes*

  - Dependencies (does it add or upgrade a dependency): (yes / no)
  - The public API: (yes / no)
  - The schema: (yes / no / don't know)
  - The default values of configurations: (yes / no)
  - The wire protocol: (yes / no)
  - The rest endpoints: (yes / no)
  - The admin cli options: (yes / no)
  - Anything that affects deployment: (yes / no / don't know)

### Documentation

Check the box below or label this PR directly.

Need to update docs? 

- [X] `doc-required` 
(Your PR needs to update docs and you will update later)
  
- [ ] `doc-not-needed` 
(Please explain why)
  
- [ ] `doc` 
(Your PR contains doc changes)

- [ ] `doc-complete`
(Docs have been already added)

### CODE REVIEW

**Review by liudezhi2098 — APPROVED** [review]

**Review by freeznet — APPROVED** [review]

**Comment by liudezhi2098** [comment]
@nlu90 help handle Conflicting files.

File: pulsar-functions/proto/src/main/proto/Function.proto [hunk 1]
```diff
@@ -165,6 +165,7 @@ message SourceSpec {
     bool cleanupSubscription = 11;
     SubscriptionPosition subscriptionPosition = 12;
     uint64 negativeAckRedeliveryDelayMs = 13;
+    bool skipToLatest = 14;
```
Comment (michaeljmarshall) [inline pulsar-functions/proto/src/main/proto/Function.proto:168]: @nlu90 - was this discussed on the mailing list? This is a protobuf change. We have a general rule that those go through the PIP process. While there could be exceptions, we should always post about these kinds of changes on the mailing list.

**Comment by momo-jun** [comment]
Hi @nlu90, I'm checking in to follow up with the doc updates. Did you have any plans to update the docs?

**Comment by nlu90** [comment]
> Hi @nlu90, I'm checking in to follow up with the doc updates. Did you have any plans to update the docs?

@momo-jun Let me know where I can update the docs.

File: pulsar-functions/proto/src/main/proto/Function.proto (hunk 1 above)
Comment (nlu90) [reply pulsar-functions/proto/src/main/proto/Function.proto:168]: @michaeljmarshall As we discussed via slack, I'll email the mailing list about the function proto change.
But it should be backward compatible and won't cause any issue for previous users.

**Comment by momo-jun** [comment]
@nlu90 shall we add `skipToLatest` to [here](https://pulsar.apache.org/docs/next/functions-cli/#consumerconfig)? Maybe @freeznet @liudezhi2098 can also advise.
BTW, the pulsar docs have been moved to the [site repo](https://github.com/apache/pulsar-site/tree/main/docs).

**Pull request merged**